openpathsampling/movechange.py

```diff
@@ -365,6 +365,26 @@ def _get_results(self):
         return []
 
 
+class RejectedNaNSampleMoveChange(RejectedSampleMoveChange):
+    """
+    Represents an rejected SamplePMC because of occurance of NaN
+
+    This will return no samples also as its result, hence it is
+    rejected.
+    """
+    pass
+
+
+class RejectedMaxLengthSampleMoveChange(RejectedSampleMoveChange):
+    """
+    Represents an rejected SamplePMC because of occurance of NaN
```

Comment:
Thanks for the catch. Yes, they return the last valid sub trajectory without the broken frame. Reason is mostly because some platforms do not even return the broken snapshot, while some return one inclusing nan.